### ISSUE TITLE
Download request matching no data - ReturnedData.as_dataframe() raises TypeError

### ISSUE THREAD

**Issue by lmar76** [ESA-VirES/VirES-Python-Client#45]
After implementing fix of https://github.com/ESA-VirES/WebClient-Framework/issues/385 is now possible to send requests also in case of missing data in the requested time interval. In this case the server returns no data. The conversion of the returned data to xarray returns `None` however the conversion to pandas dataframe raises a `TypeError`:
```sh
In [1]: from viresclient import SwarmRequest                                                                                                                                                              

In [2]: request = SwarmRequest()                                                                                                                                                                          

In [3]: request.set_collection('SW_OPER_MAGA_LR_1B')                                                                                                                                                      
Out[3]: <viresclient._client_swarm.SwarmRequest at 0x7f02b5eb3be0>

In [4]: request.set_products(measurements=request.available_measurements('MAG'))                                                                                                                          
Out[4]: <viresclient._client_swarm.SwarmRequest at 0x7f02b5eb3be0>

In [5]: data = request.get_between('2014-01-05T00:00:00', '2014-01-05T01:00:00')                                                                                                                          
[1/1] Processing:  100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|  [ Elapsed: 00:01, Remaining: 00:00 ]
      Downloading: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|  [ Elapsed: 00:00, Remaining: 00:00 ] (0.008MB)

In [6]: data.to_file('test.cdf')                                                                                                                                                                          
Data written to test.cdf

In [7]: xr = data.as_xarray()                                                                                                                                                                             
Warning:  Unable to create dataset from part 1 of 1 
(This part is likely empty)

In [8]: type(xr)                                                                                                                                                                                          
Out[8]: NoneType

In [9]: df = data.as_dataframe()                                                                                                                                                                          
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-db0dbceb1c60> in <module>
----> 1 df = data.as_dataframe()

/opt/conda/lib/python3.7/site-packages/viresclient/_data_handling.py in as_dataframe(self, expand)
    507         """
    508         return pandas.concat(
--> 509             [d.as_dataframe(expand=expand) for d in self.contents])
    510 
    511     def as_xarray(self):

/opt/conda/lib/python3.7/site-packages/viresclient/_data_handling.py in <listcomp>(.0)
    507         """
    508         return pandas.concat(
--> 509             [d.as_dataframe(expand=expand) for d in self.contents])
    510 
    511     def as_xarray(self):

/opt/conda/lib/python3.7/site-packages/viresclient/_data_handling.py in as_dataframe(self, expand)
    368             df = make_pandas_DataFrame_from_csv(self._file.name)
    369         elif self.filetype == 'cdf':
--> 370             df = make_pandas_DataFrame_from_cdf(self._file.name, expand=expand)
    371         return df
    372 

/opt/conda/lib/python3.7/site-packages/viresclient/_data_handling.py in make_pandas_DataFrame_from_cdf(cdf_filename, expand)
    176                     df[col + suffix] = cdf.varget(col)[:, component]
    177             else:
--> 178                 df[col] = list(cdf.varget(col))
    179         # Convert timestamps to datetime objects
    180         df.index = pandas.to_datetime(

TypeError: 'NoneType' object is not iterable
```

**Comment by smithara** [comment]
I found that this is broken between cdflib 0.3.16 and 0.3.17. Investigating further

I'm also wondering if the expected behaviour with empty datasets should be to return `None` or a dataframe/dataset containing the variable names but empty data.

**Comment by pacesm** [comment]
I opt for the latter
> a dataframe/dataset containing the variable names but empty data.

because that is what you get from the server.

**Comment by lmar76** [comment]
I agree with @pacesm

**Comment by smithara** [comment]
I just released v0.5.0 (not yet on VRE) where this is fixed, along with returning empty dataframes/datasets in these cases.

The source of the error was a change in behaviour in cdflib, where in 0.3.16, `cdf.varget(x)` returned `ValueError: No data is written for this variable`, in 0.3.17 it returns `None`. I refactored my code a bit to make it easier to identify such upstream changes in the future, or maybe to drop in an alternative package to replace it.
